configuration/pih/htmlforms/hiv/section-hiv-state.xml

```diff
@@ -156,10 +156,248 @@
+
+         let lastRecordedInhEndDate = {
+            endDate: "",
+            date: null,
+            encounterDisplay: ""
+        };
+
+        
+           jq.getJSON(apiBaseUrl + "/obs", {
+            concept: inhEndateCOnceptUuid,
+            patient: patientUuid,
+            v: 'custom:(uuid,display,obsDatetime,value,encounter:(display,encounterDatetime,location:(display),encounterType:(display)))'
+          },
+          function( data ){
+            if (data.results.length > 0) {
+                for (let index = 0; index &lt; data.results.length; ++index) {
+                    let inhEndDateObs = data.results[index];
+                    let inhEndDateObsDate = new Date(inhEndDateObs.obsDatetime);
+                    let tmpInhEndDateObs = {
+                      endDate: "",
+                      date: null,
+                      encounterDisplay: ""
+                    };
+                   
+
+                    if( (currentEncounterDate.getTime() >=  inhEndDateObsDate.getTime()) ||
+                        (currentEncounterDate.toDateString() === inhEndDateObsDate.toDateString())) {
+                            if (typeof inhEndDateObs.value !== "undefined" &amp;&amp; inhEndDateObs.value !== null &amp;&amp; (inhEndDateObs.value.length &gt; 0)) {
+                                tmpInhEndDateObs.endDate = inhEndDateObs.value;
+                                tmpInhEndDateObs.date = inhEndDateObsDate;
+                                tmpInhEndDateObs.encounterDisplay = inhEndDateObs.encounter.display;
+
+                                
+                              }
+                    }
+                    if ( (tmpInhEndDateObs.date != null) &amp;&amp;
+                        ((lastRecordedInhEndDate.date == null) || (lastRecordedInhEndDate.date.getTime() &lt; tmpInhEndDateObs.date.getTime()) ) ) {
+
+                        lastRecordedInhEndDate.endDate = tmpInhEndDateObs.endDate;
+                        lastRecordedInhEndDate.date = tmpInhEndDateObs.date;
+                        lastRecordedInhEndDate.encounterDisplay = tmpInhEndDateObs.encounterDisplay;
+                    }
+                }
+            }
+            if (lastRecordedInhEndDate.date != null) {
+                  let domEl = getField('inhEndDate.value')
+                   domEl.datepicker("setDate", new Date(lastRecordedInhEndDate.endDate));
+                  }
+          });
     });
 </script>
 </excludeIf>
 
+<!-- This method allows you to retrieve the last start date of izoniazid -->
+<excludeIf velocityTest="$fn.getObs($encounter, 'PIH:14534')">
+<script type="text/javascript">
+
+    jq(document).ready(function() {
+        
+
+        const contextPath = window.location.href.split('/')[3];
+        const apiBaseUrl =  "/" + contextPath + "/ws/rest/v1";
+        const patientUuid = '<lookup expression="patient.uuid"/>';
+
+        let currentEncounterDate = new Date();
```

Comment:
Current encounter date shouldn't default to "today" if the form field isn't present, but rather it should default to the current encounter date of the existing encounter (if any) via velocity expression.